### PR TITLE
Improve duration declaration

### PR DESCRIPTION
**Pull request by godwhoa**
TODO:

- [x] Port more options to use duration
- [x] Config option docs
- [x] Manual testing
- [x] Vendor pytimeparse
- [x] Add more unittests

Porting Status:
- [x] respawn_time
- [x] grief_friendly_fire_time
- [x] teamswitch_interval
- [x] squads.respawn_time
- [x] votemap.extension_time
- [x] votemap.time
- [x] spawn_protect.protection_time
- [x] spectator_control.kick_time
- [x] bans.default_duration
- [x] afk.time_limit
- [x] votekick.ban_duration
- [x] antijerk.ban_duration
- [x] default_time_limit

Closes: #416 
Closes: #330

### CODE REVIEW

**Comment by NotAFile** [comment]
not sure I'm a fan of vendoring it like this... can't we just take the file, the license text, and put it into util.py or something?

**Comment by godwhoa** [comment]
@NotAFile Sure

**Comment by godwhoa** [comment]
Commit history looks noisy should I try and squash few groups of commits together?

File: doc/config-options.rst [hunk 1]
```diff
@@ -198,10 +198,10 @@ a block when :ref:`friendly_fire_on_grief` is enabled.
 teamswitch_interval
 +++++++++++++++++++
 
-Forces players to wait a number of minutes before being able to switch back
+Forces players to wait a set duration before being able to switch back
 again after they switched teams.
 
-0 disables the cooldown.
+"0sec" disables the cooldown.
```
Comment (NotAFile) [inline doc/config-options.rst]: hm, I'd love if just a pain "0" still worked. After all, it doesn't matter if it's years or months if it's zero.

Comment (godwhoa) [reply doc/config-options.rst]: That'd work actually. If you pass it digits it'll take it as seconds. https://github.com/piqueserver/piqueserver/blob/6c7eadcb86ebe4a0ebf05d1427fc5c3ef9a1e6d4/piqueserver/config.py#L294-L300

Comment (godwhoa) [reply doc/config-options.rst]: For the sake of consistence made everything use duration.

**Comment by NotAFile** [comment]
Any reason why this hasn't this been merged yet?

**Comment by godwhoa** [comment]
@NotAFile Nope, good to merge.

**Comment by NotAFile** [comment]
This commit broke stuff because `piqueserver/utils` wasn't declared in the setup.py. I've pushed a fix, but why didn't CI catch this?

**Pull request merged**